vegbilder-client/src/components/ImageView/ImageView.tsx

```diff
@@ -34,14 +34,127 @@ interface IImageViewProps {
+    switch (action.type) {
+      case 'clientX': {
+        return {
+          ...state,
+          clientX: action.newVal,
+        };
+      }
+      case 'clientY': {
+        return {
+          ...state,
+          clientY: action.newVal,
+        };
+      }
+      case 'scrollX': {
+        return {
+          ...state,
+          scrollX: state.scrollX + action.newVal - state.clientX,
+          clientX: action.newVal,
+        };
+      }
+      case 'scrollY': {
+        return {
+          ...state,
+          scrollY: state.scrollY + action.newVal - state.clientY,
+          clientY: action.newVal,
+        };
+      }
+      default:
+        return state;
+    }
+  };
+
+  const [scrollState, dispatch] = useReducer(scrollReducer, initialScrollState);
+
+  // We add mouse event handlers that lets the user drag the image to scroll. If the user only clicks we zoom in/out.
+  useEffect(() => {
+    const currentImageContainerRef = imageContainerRef.current;
+    if (!currentImageContainerRef) return;
+
+    let shouldScroll = false;
+    let mouseMoved = false;
+
+    const onMouseDown = (event: MouseEvent) => {
+      event.preventDefault();
+      dispatch({ type: 'clientX', newVal: event.clientX });
+      dispatch({ type: 'clientY', newVal: event.clientY });
+      shouldScroll = true;
+      mouseMoved = false;
+    };
+
+    const onMouseUp = () => {
+      shouldScroll = false;
+      setCursor('zoom-out');
+      if (!mouseMoved) setIsZoomedInImage((prevState) => !prevState);
+    };
+
+    const onMouseOut = () => {
+      shouldScroll = false;
+    };
+
+    const onMouseMove = (event: MouseEvent) => {
+      event.preventDefault();
+      mouseMoved = true;
+      if (shouldScroll) {
+        setCursor('grab');
+        if (currentImageContainerRef) {
+          currentImageContainerRef.scrollLeft =
+            scrollState.scrollX + event.clientX - scrollState.clientX;
+          currentImageContainerRef.scrollTop =
+            scrollState.scrollY + event.clientY - scrollState.clientY;
+        }
+        dispatch({ type: 'scrollX', newVal: event.clientX });
+        dispatch({ type: 'scrollY', newVal: event.clientY });
+      }
+    };
+
+    currentImageContainerRef.addEventListener('mousedown', (event) => onMouseDown(event));
+    currentImageContainerRef.addEventListener('mouseup', () => onMouseUp());
+    currentImageContainerRef.addEventListener('mouseout', () => onMouseOut());
+    currentImageContainerRef.addEventListener('mousemove', (event) => onMouseMove(event));
+
+    return () => {
+      currentImageContainerRef.removeEventListener('mousedown', (event) => onMouseDown(event));
+      currentImageContainerRef.removeEventListener('mouseup', () => onMouseUp());
+      currentImageContainerRef.removeEventListener('mouseout', () => onMouseOut());
+      currentImageContainerRef.removeEventListener('mousemove', (event) => onMouseMove(event));
+    };
+  }, [imageContainerRef]);
```

Comment:
Burde du egentlig ha `imageContinerRef.current` i dependency-lista? 🤔  Jeg ville trodd at det her kun blir rendret én gang, ved første render, ettersom et referanseobjekt fra `useRef` er stabilt og alltid vil være det samme objektet  for en gitt komponent.